dips/dip-1.mdx

```diff
@@ -211,18 +211,88 @@ Represents an error that occurred in response to a Command.
+
+`unknown_field`: field is unknown for an object.
+
+`unknown_command_type`: invalid/unsupported `command_type`.
+
+`invalid_field_value`:
+* Invalid / unknown enum field values.
+* UUID field value does not match UUID format.
+* Payment actor address is not a valid DIP-5 account identifier.
+* Currency field value is not a valid Diem currency code for the connected network.
+
+`invalid_command_producer`: The HTTP request sender is not the right actor to send the payment object. For example, if the actor receiver sends a new command with payment object change that should be done by actor sender.
+
+`invalid_initial_or_prior_not_found`: could not find command by reference_id for a non-initial state command object; for example, actor receiver received a payment command object that actor sender status is `ready_for_settlement`, but receiver could not find any command object by the reference id.
+
+`no_kyc_needed`: payment action amount is under travel rule limit.
+
+`invalid_recipient_signature`:
+* Field `recipient_signature` value is not hex-encoded bytes.
+* Field `recipient_signature` value is an invalid signature.
+
+`unknown_address`:
+* The DIP-5 account identifier address in the command object is not HTTP request sender’s address or receiver’s address. For payment object it is `sender#address` or `receiver#address`.
+* Could not find on-chain account by an DIP-5 account identifier address in command object address.
+
+`conflict`:
+* Command object is in conflict with another different command object by cid, likely a cid is reused for different command object.
+* Failed to acquire lock by for the command object by the `reference_id`.
+
+`unsupported_currency`: field `payment#action#currency` value is a valid Diem currency code, but it is not supported / acceptable by the receiver VASP.
+
+`missing_dependency`: One of dependency objects is not found by the id provided in the command object. For example, the receiver should return this error code if it could not find the data by the `original_payment_reference_id` if the sender set it.
+
+`invalid_dependency`: One of dependency objects is found, but it is not valid for the command object. For example, the dependency payment object status is `aborted` instead of `ready_for_settlement`.
+
+`invalid_overwrite`:
+* Overwrite a field that can only be written once.
+* Overwrite an immutable field (field can only be set in initial command object), e.g. `original_payment_reference_id`).
+* Overwrite opponent payment actor's fields.
+
+`invalid_transition`: As we only allow one actor action at a time, and the next states for a given command object state are limited to specific states. This error indicates the new payment object state is not valid according to the current object state. For example: VASP A sends RSOFT to VASP B, VASP B should send the next payment object with ABORT, or SSOFTSEND; VASP A should respond to this error code if VASP B sends payment object state SSOFT.
```

Comment:
I think this actually covers `invalid_command_producer`

Reply:
it is not, a command object with invalid_command_producer can be valid transition. An invalid_command_producer is more of validating the command object with the x_request_sender_address.

Reply:
> a command object with invalid_command_producer can be valid transition

can you give an example? Note that only the sender can change sender's status, same to receiver.

Reply:
Let's say sender sends SINT, then sender sends another new command RSEND, with all information set for receiver. It changes receiver actor object, which is invalid-overwrite-error, but it is also invalid_command_producer, and the result is a valid transition.

It probably related to what is a scope of "transition", from my point view, it is from command object A to B, regardless of who created B from A. But if you consider what actor create B from A is also part of transition validation, then yes, it is covered.
It's common some error codes may cover each other in a complex scenario, we don't need judge what and when must be used.
For example, technically json `"{}" ` can trigger a lot of errors, like invalid transition, invalid overwrite, missing fields..... Application just need try to return reasonable one for it.

Reply:
> Let's say sender sends SINT, then sender sends another new command RSEND, with all information set for receiver. It changes receiver actor object, which is invalid-overwrite-error, but it is also invalid_command_producer, and the result is a valid transition.

SENDER shouldn't be allowed to send a `RSEND` command and it can be interpreted from two different aspects:
1. to get to RSEND, sender needs to touch receiver's data, this is not allowed (they shall only update their own data, respectively)
2. SENDER should never reach RSEND, because it is a state that only receiver can move into.

But this is fair enough, no strong opinion over one or the other

Reply:
You're right about sender should never reach RSEND, I'm giving an example of invalid/buggy implementation that may give "a command object with invalid_command_producer can be valid transition"